src/Adapter/MSTest.TestAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -199,6 +199,13 @@ internal UnitTestElement GetTestFromMethod(MethodInfo method, bool isDeclaredInT
         var testMethodAttribute = _reflectHelper.GetCustomAttribute<TestMethodAttribute>(method);
         testElement.DisplayName = testMethodAttribute?.DisplayName ?? method.Name;
 
+        // OR get DisplayName from DataTestMethodAttribute
+        if (testMethodAttribute == null)
+        {
+            var dataTestMethodAttribute = _reflectHelper.GetCustomAttribute<DataTestMethodAttribute>(method);
+            testElement.DisplayName = dataTestMethodAttribute?.DisplayName ?? method.Name;
+        }
```

Comment:
I will run some test locally but this part should not be required. As far as I remember `GetCustomAttribute` is looking for inherited attributes so should take `DataTestMethod` or any custom child you would create.

Reply:
You're probably right. It's actually a flag that you can provide: https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=net-8.0#system-reflection-memberinfo-getcustomattributes(system-type-system-boolean)
The issue here is the mocked ReflectHelper, you can only Setup the call for a certain attribute, it isn't actually doing the reflection call. So in a real scenario it will will work, but it would not be possible to write a test to verify this, or at least not at the point where I added the test. (I'm not familiar with all the code so you may have a better suggestion)

Reply:
Sorry got busy with quite a few things. I was able to confirm that this part is not needed and that a manual test confirms this works. I also saw that we are using mocks who are causing issue in the unit test. I'll have a look during the weekend or early next week.

Reply:
Ok I did a quick (dirty change) this is not ideal test but I don't think the mocking is really best test we can do. As we are super close to release date, I will ship as-is and later I will work on adding some integration test for that.